aws/resource_aws_cloud9_environment_ec2_test.go

```diff
@@ -0,0 +1,237 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloud9"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSCloud9EnvironmentEc2_basic(t *testing.T) {
+	var conf cloud9.Environment
+
+	rString := acctest.RandString(8)
+	envName := fmt.Sprintf("tf_acc_env_basic_%s", rString)
+	uEnvName := fmt.Sprintf("tf_acc_env_basic_updated_%s", rString)
+
+	resourceName := "aws_cloud9_environment_ec2.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSCloud9EnvironmentEc2Destroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccAWSCloud9EnvironmentEc2Config(envName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSCloud9EnvironmentEc2Exists(resourceName, &conf),
+					resource.TestCheckResourceAttr(resourceName, "instance_type", "t2.micro"),
+					resource.TestCheckResourceAttr(resourceName, "name", envName),
+					resource.TestCheckResourceAttrSet(resourceName, "arn"),
```

Comment:
Minor nitpick: We can regexp match the ARN according to the [ARN Cloud9 documentation](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#arn-syntax-cloud9):

```go
resource.TestMatchResourceAttr(resourceName, "arn", regexp.MustCompile(fmt.Sprintf("^arn:[^:]+:cloud9:[^:]+:[^:]+:environment:.+$"))
```